crates/rpc/src/v02/method/add_declare_transaction.rs

```diff
@@ -63,8 +65,35 @@ pub async fn add_declare_transaction(
                     tx.max_fee,
                     tx.signature,
                     tx.nonce,
-                    contract_definition,
+                    ContractDefinition::Cairo(contract_definition),
                     tx.sender_address,
+                    None,
+                    input.token,
+                )
+                .await?;
+
+            Ok(AddDeclareTransactionOutput {
+                transaction_hash: response.transaction_hash,
+                class_hash: response.class_hash,
+            })
+        }
+        // TODO: Verify compiled class hash (as an additional check)
```

Comment:
I'm not sure we need that. I mean we probably should do a check in `starknet_estimateFee`, because otherwise you won't get an error message if the compiled class hash doesn't match.

Here I guess the gateway does a check anyway, so you'll get an error message in case of a class hash mismatch.

Reply:
I added a missing error variant for that occasion (`StarknetErrorCode.INVALID_COMPILED_CLASS_HASH`).